test/rest_api/test_task_manager.py

```diff
@@ -0,0 +1,123 @@
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+            args1 = { "sequence_number": 2, "keyspace": "keyspace0", "table": "table1"}
+            with new_test_task(rest_api, args1) as task1:
+                print(f"created test task {task1}")
+                resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+                resp.raise_for_status()
+                tasks = [task0, task1]
+                for task in resp.json():
+                    task_id = task["task_id"]
+                    assert task_id in tasks, f"Unrecognized task_id={task_id}"
+                    tasks.remove(task_id)
+                assert not tasks, f"list_module_tasks did not return all tasks. remaining={tasks}"
+
+def test_task_manager_status_running(rest_api):
+    with new_test_module(rest_api):
+        args0 = { "sequence_number": 1, "keyspace": "keyspace0", "table": "table0"}
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+            resp = rest_api.send("GET", f"task_manager/task_status/{task0}")
+            resp.raise_for_status()
+
+            status = resp.json()
+            assert status["id"] == task0, "Incorrect task id"
+            assert status["state"] == "running", "Incorrect task state"
+            assert status["sequence_number"] == 1, "Incorrect task sequence number"
+            assert status["keyspace"] == "keyspace0", "Incorrect task keyspace"
+            assert status["table"] == "table0", "Incorrect task table"
+
+            resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+            tasks = resp.json()
+            assert tasks, "task_status unregistered task that did not finish"
+
+def test_task_manager_status_done(rest_api):
+    with new_test_module(rest_api):
+        args0 = { "sequence_number": 1, "keyspace": "keyspace0", "table": "table0"}
+        with new_test_task(rest_api, args0) as task0:
+            print(f"created test task {task0}")
+
+            resp = rest_api.send("POST", f"task_manager_test/finish_test_task/{task0}")
+            resp.raise_for_status()
+            resp = rest_api.send("GET", f"task_manager/task_status/{task0}")
+            resp.raise_for_status()
+
+            status = resp.json()
+            assert status["id"] == task0, "Incorrect task id"
+            assert status["state"] == "done", "Incorrect task state"
+            assert status["error"] == "", "Error message was set"
+            assert status["sequence_number"] == 1, "Incorrect task sequence number"
+            assert status["keyspace"] == "keyspace0", "Incorrect task keyspace"
+            assert status["table"] == "table0", "Incorrect task table"
+
+            resp = rest_api.send("GET", "task_manager/list_module_tasks/test")
+            tasks = resp.json()
+            assert not tasks, "task_status did not unregistered task that finished"
```

Comment:
Please add a test for the task ttl that the tasks disappears after ttl+1 seconds